src/main/sensors/compass.c

```diff
@@ -388,31 +404,245 @@ uint32_t compassUpdate(timeUs_t currentTimeUs)
     }
 
     flightDynamicsTrims_t *magZero = &compassConfigMutable()->magZero;
-    if (magInit) {              // we apply offset only once mag calibration is done
-        mag.magADC[X] -= magZero->raw[X];
-        mag.magADC[Y] -= magZero->raw[Y];
-        mag.magADC[Z] -= magZero->raw[Z];
-    }
 
     if (tCal != 0) {
-        if ((currentTimeUs - tCal) < 30000000) {    // 30s: you have 30s to turn the multi in all directions
+        if ((currentTimeUs - (tCal + movementStartedTimeUs)) < CALIBRATION_TIME_US) {
```

Comment:
You are summing tCAl (micros at compassStartCalibration) with movementStartedTimeUs (micros movement started). Slightly more complicated logic is necessary. Maybe just use tCal as timeout first and reset it on first movement

Reply:
Nice, will do.